dms/models/ir_attachment.py

```diff
@@ -0,0 +1,64 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    def _get_dms_directories(self, res_model, res_id):
+        return self.env["dms.directory"].search(
+            [("res_model", "=", res_model), ("res_id", "=", res_id)]
+        )
+
+    def _dms_operations(self, attachment_ids):
+        for attachment in attachment_ids:
+            if attachment.res_model and attachment.res_id:
+                directories = self._get_dms_directories(
+                    attachment.res_model, attachment.res_id
+                )
+                if not directories:
+                    # Auto-create directories
+                    directories_without_res_id = self._get_dms_directories(
```

Comment:
Maybe it's better to extract this part to a method.

Reply:
Yes, it's a good idea, separated now.